### PR TITLE
Add basic-auth proxy

### PR DESCRIPTION
**Pull request by bgpat**
## WHY

Proxy with Basic-Authentication for publishing private godoc.

## WHAT

Use [nginx-basic-auth-proxy](https://github.com/dtan4/nginx-basic-auth-proxy).

### CODE REVIEW

**Comment by dtan4** [comment]
You don't have to prepare new Deployment for proxy. basic-auth-proxy container can be included in the same Pod as godoc server.

Please see https://github.com/wantedly/aws-es-auth-proxy/blob/master/kubernetes/deployment.yaml which has basic-auth-proxy and aws-sign-proxy in the same Pod.

**Comment by bgpat** [comment]
Fixed https://github.com/wantedly/godoc/pull/4#issuecomment-284951827 and changed port for basic-auth proxy.

File: kubernetes/server-svc.yaml [hunk 1]
```diff
@@ -9,9 +9,9 @@ metadata:
 spec:
   ports:
   - name: http
-    port: 80
-    targetPort: 8080
+    port: 443
+    targetPort: 8081
   selector:
     name: server
```
Comment (dtan4) [inline kubernetes/server-svc.yaml]: Please rename as `nginx-basic-auth-proxy`.

File: kubernetes/server.yaml [hunk 2]
```diff
@@ -33,3 +31,24 @@ spec:
...
+          valueFrom:
+            secretKeyRef:
+              name: dotenv
+              key: nginx-basic-auth-password
+        - name: PROXY_PASS
+          value: http://localhost:8080/
+        - name: PORT
+          value: "8081"
+        - name: SERVER_NAME
+          value: godoc.wantedly.com
```
Comment (dtan4) [inline kubernetes/server.yaml]: This repository is public, so `SERVER_NAME` should also be set from Kubernetes secret, as well as `BASIC_AUTH_*`.

Comment (bgpat) [inline kubernetes/server.yaml]: Fixed to use secret.

File: kubernetes/server-svc.yaml (hunk 1 above)
Comment (bgpat) [inline kubernetes/server-svc.yaml]: I think I should rename https://github.com/wantedly/godoc/blob/733190ae1a18b641241c8cfc6189874300a29f19/kubernetes/server.yaml#L4 also.
It didn't work in minikube.

Comment (dtan4) [inline kubernetes/server-svc.yaml]: Sorry, I was wrong 😝 It's okay...

File: kubernetes/server-svc.yaml [hunk 3]
```diff
@@ -9,9 +9,9 @@ metadata:
...
     role: web
-type: LoadBalancer
+  type: NodePort
```
Comment (dtan4) [inline kubernetes/server-svc.yaml]: To deploy to AWS k8s, this should be `LoadBalancer`, shouldn't it?

Comment (bgpat) [inline kubernetes/server-svc.yaml]: Yes, service type should be `LoadBalancer`.
I fixed it :bow:

**Comment by dtan4** [comment]
LGTM 👍

**Pull request merged**